### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by cnasikas**
- [x] Show user statistics
- [x] Better error messages
- [x] CLI

### CODE REVIEW

**Comment by dionyziz** [comment]
Push these and let's move forward. We can do a review together later.

**Comment by cnasikas** [comment]
Ok!

File: README.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ Assignments by default are assumed to be inside `db/assignments`. You can change
...
 
-Each assignment has to implement `judge (aux, user, assignment, solution)` and `aux (user, assignment)` functions. `judge` function takes as input an `aux`, a `user`, an `assignment` and a `solution`, and returns `1` in case of success and `0` otherwise. The `aux` function takes as input the `user` and the `assignment` and return an aux `{ private: 'privateAux', public: 'publicAux' }` where `private` and `public` can be of any type and can be omitted.
+Each assignment has to implement `judge (aux, user, assignment, solution)` and `aux (user, assignment)` functions. `judge` function takes as input an `aux`, a `user`, an `assignment` and a `solution`, and returns and object containing the grade and a message to show to the user. The return object's format must be: `{ grade: 0, msg: 'A message' }` The `aux` function takes as input the `user` and the `assignment` and return an aux `{ private: 'privateAux', public: 'publicAux' }` where `private` and `public` can be of any type and can be omitted.
```
Comment (OrfeasLitos) [inline README.md]: returns and object -> returns an object
and return an aux -> and returns an aux
`{ private: 'privateAux', public: 'publicAux' }` -> `{ private: someObj, public: anotherObj }` (don't use strings)

Comment (cnasikas) [reply README.md]: Thanks!

**Comment by cnasikas** [comment]
The CLI refactor allows to run exercises without the need of api and app. You can generate aux and judge your assignments. I hope this feature would speed up the process of assignment creation and make it easier for you guys. Please test it!!

Examples:

```
node cli.js generate aux proof-of-work
node cli.js generate aux proof-of-work --user.id=2 --user.username='foo'
node cli.js judge proof-of-work --solution 'a_solution' --aux.public=2
node cli.js judge simple-storage --solution=/Users/cnasikas/workspace/solidity/SimpleStorage.sol --file
```

**Comment by OrfeasLitos** [comment]
cli returns error either way:
```
$ node cli/cli.js generate aux db/assignments/proof_of_work 
sequelize deprecated String based operators are now deprecated. Please use Symbol based operators for better security, read more at http://docs.sequelizejs.com/manual/tutorial/querying.html#operators db/node_modules/sequelize/lib/sequelize.js:242:13
2037036
info: Skipping non-assignment {"file":"/blabla/repositories/blockchain-course/db/assignments/proof_of_work/prove.js","meta":{"file":"/blabla/repositories/blockchain-course/db/assignments/proof_of_work/prove.js"}}
An error has occured: Resource not found
➜  blockchain-course git:(pr) ✗ node cli/cli.js generate aux proof_of_work
sequelize deprecated String based operators are now deprecated. Please use Symbol based operators for better security, read more at http://docs.sequelizejs.com/manual/tutorial/querying.html#operators db/node_modules/sequelize/lib/sequelize.js:242:13
2037036
info: Skipping non-assignment {"file":"/blabla/repositories/blockchain-course/db/assignments/proof_of_work/prove.js","meta":{"file":"/blabla/repositories/blockchain-course/db/assignments/proof_of_work/prove.js"}}
An error has occured: Resource not found
```

**Comment by cnasikas** [comment]
> cli returns error either way:
> 
> ```
> $ node cli/cli.js generate aux db/assignments/proof_of_work 
> sequelize deprecated String based operators are now deprecated. Please use Symbol based operators for better security, read more at http://docs.sequelizejs.com/manual/tutorial/querying.html#operators db/node_modules/sequelize/lib/sequelize.js:242:13
> 2037036
> info: Skipping non-assignment {"file":"/blabla/repositories/blockchain-course/db/assignments/proof_of_work/prove.js","meta":{"file":"/blabla/repositories/blockchain-course/db/assignments/proof_of_work/prove.js"}}
> An error has occured: Resource not found
> ➜  blockchain-course git:(pr) ✗ node cli/cli.js generate aux proof_of_work
> sequelize deprecated String based operators are now deprecated. Please use Symbol based operators for better security, read more at http://docs.sequelizejs.com/manual/tutorial/querying.html#operators db/node_modules/sequelize/lib/sequelize.js:242:13
> 2037036
> info: Skipping non-assignment {"file":"/blabla/repositories/blockchain-course/db/assignments/proof_of_work/prove.js","meta":{"file":"/blabla/repositories/blockchain-course/db/assignments/proof_of_work/prove.js"}}
> An error has occured: Resource not found
> ```

You need to pass the ID (name) of the assignment and not the file path. That's why you get `An error has occurred: Resource not found`

The `info: Skipping non-assignment` info indicates that the class exported in `prove.js` is not an instance of `BaseJudge`. Would you mind posting `prove.js` 

About `2037036` I can't reproduce it. Seems as a solution of proof-of-work is printed somewhere in your code.

**Comment by cnasikas** [comment]
@dionyziz @OrfeasLitos I am thinking to merge. Because we are in rush I think is better to develop fast with new features and maybe small bugs than the opposite. What do you think ?

**Comment by OrfeasLitos** [comment]
I don't want to drag you behind, so I'm OK with the risk of merging before reviewing.

**Comment by dionyziz** [comment]
Yes

**Pull request merged**